test/src/main/java/org/apache/accumulo/test/functional/TabletMergeabilityIT.java

```diff
@@ -0,0 +1,219 @@
+
+    }
+  }
+
+  @Test
+  public void testMergeabilityMultipleRanges() throws Exception {
+    String tableName = getUniqueNames(1)[0];
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      c.tableOperations().create(tableName);
+      var tableId = TableId.of(c.tableOperations().tableIdMap().get(tableName));
+
+      SortedMap<Text,TabletMergeability> splits = new TreeMap<>();
+      splits.put(new Text(String.format("%09d", 333)), TabletMergeability.never());
+      splits.put(new Text(String.format("%09d", 555)), TabletMergeability.never());
+      splits.put(new Text(String.format("%09d", 666)), TabletMergeability.never());
+      splits.put(new Text(String.format("%09d", 999)), TabletMergeability.never());
+
+      c.tableOperations().putSplits(tableName, splits);
+      Wait.waitFor(() -> countTablets(getCluster().getServerContext(), tableName, tm -> true) == 5,
+          5000, 500);
+
+      splits.put(new Text(String.format("%09d", 333)), TabletMergeability.always());
+      splits.put(new Text(String.format("%09d", 555)), TabletMergeability.always());
+      // Keep tablet 666 as never, this should cause two fate jobs for merging
+      splits.put(new Text(String.format("%09d", 999)), TabletMergeability.always());
+      c.tableOperations().putSplits(tableName, splits);
+
+      // Wait for merge, we should have 3 tablets
+      // 333 and 555 should be merged into 555
+      // 666
+      // 999 and default merged into default
+      Wait.waitFor(() -> hasExactTablets(getCluster().getServerContext(), tableId,
+          Set.of(new KeyExtent(tableId, new Text(String.format("%09d", 555)), null),
+              new KeyExtent(tableId, new Text(String.format("%09d", 666)),
+                  new Text(String.format("%09d", 555))),
+              new KeyExtent(tableId, null, new Text(String.format("%09d", 666))))),
+          10000, 200);
+
+    }
+  }
+
+  @Test
+  public void testSplitAndMergeAll() throws Exception {
+    String tableName = getUniqueNames(1)[0];
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+      Map<String,String> props = new HashMap<>();
+      props.put(Property.TABLE_SPLIT_THRESHOLD.getKey(), "32K");
+      props.put(Property.TABLE_FILE_COMPRESSED_BLOCK_SIZE.getKey(), "1K");
+      c.tableOperations().create(tableName, new NewTableConfiguration().setProperties(props));
+      var tableId = TableId.of(c.tableOperations().tableIdMap().get(tableName));
+
+      // Ingest data so tablet will split
+      VerifyParams params = new VerifyParams(getClientProps(), tableName, 50_000);
+      TestIngest.ingest(c, params);
+      VerifyIngest.verifyIngest(c, params);
+
+      // Wait for table to split, should be more than 10 tablets
+      Wait.waitFor(() -> c.tableOperations().listSplits(tableName).size() > 10, 10000, 200);
+
+      // Delete all the data
+      c.tableOperations().deleteRows(tableName, null, null);
```

Comment:
deleteRows also merges tablets I think, so the splits may not be going away because of auto merge here.  Could delete all the data and then compact to process all the delete markers.


```suggestion
       var bd = c.createBatchDeleter(tableName, Authorizations.EMPTY, 1);
      bd.setRanges(List.of(new Range()));
      bd.delete();
      c.tableOperations().compact(tableName, new CompactionConfig().setFlush(true));
```

Reply:
Ah that is a good point, I forgot that deleteRows merges away empty tablets already so I will make that change.

Reply:
I was not sure,  I had to look at the impl.  The javadoc does not mention that delete rows will change the tables split point, its should.  Actually it does not have to change the splits, it could leave the splits alone and just drop files from tablets.  I think it does that because it was implemented using the merge tablet code.  Also in 2.1 it may add and remove splits, not sure but in 3.1 think it no longer add splits to the table and only removes.

Reply:
If I recall, I believe the reason why we use the merge code to merge away empty tablets is to preserve the behavior of previous versions of accumulo that deleted tablets. 

In older versions when we ran delete rows, the code would run a chop compaction on the tablets at the boundaries and then it could just delete the all the tablets in between the delete range.

With the addition of no-chop merges, we now have the ability to just fence files and avoid chop compactions. So now we fence the tablets on the boundaries, and can delete the metadata in between and merge away the empty tablets.

The end result in both cases is those tablets are gone, so if we do decide to keep the tablets going forward that would be a different behavior than before.

Reply:
Should we take advantage of the new TabletMergeability column? Maybe we should change deleteRows to only merge away the empty tablets if they are marked as mergeable?

Reply:
Opened #5371 based on this disucssion.